### DISCUSSION TITLE
RunTime Error: SendKeys cannot run inside this application

### DISCUSSION THREAD

**Discussion by horsto** [bonsai-rx/bonsai#1122] [Q&A]
I just transferred my workflow to a new installation of Bonsai / new setup and I am running into this (new) error message. Is this a quick fix (what is causing it?)?
 
![Capture](https://user-images.githubusercontent.com/7548928/200618230-13f7c069-3b99-4b75-ba5d-b7d3fbe09acc.PNG)
![Capture](https://user-images.githubusercontent.com/7548928/200618437-a2a25015-0cb7-4286-8632-33ecb85fbae3.PNG)

**Comment by vncntprvst** [comment]
I'm running into the same issue. The workflow below works on a given Windows computer, but not another:
![image](https://github.com/bonsai-rx/bonsai/assets/1872756/76e139de-57a3-4c3e-a493-bb00b8c02734)
The error message is the same as above: **SendKeys cannot run inside this application because the application is not handling Windows messages. Either change the application to handle messages, or use the SendKeys.SendWait method.**
Any idea?
Thanks

**Reply by glopesdev** [reply]
@vncntprvst @horsto what versions of windows are you running? Do you have a sense of what changed across the installations?

**Reply by vncntprvst** [reply]
Yeah it's likely due to different Windows version. 
Working system: desktop computer on Windows 10 Enterprise 22H2
Not working system: laptop computer on Windows 11 Home 22H2
I can switch the latter to an enterprise version and see if that resolves the issue.

**Reply by vncntprvst** [reply]
Switching to Enterprise version (Windows 11 Enterprise 22H2) did not resolve this issue, even after re-installing Bonsai. So I'm assuming it's a Windows 11 problem.

**Comment by vncntprvst** [comment]
A workaround for this issue is to use a Python script in Bonsai, such as this: 
```
import clr
clr.AddReference('System.Windows.Forms')
from System.Windows.Forms import SendKeys

def process(value):
    if value:
        # Send Ctl Shift 8 
        SendKeys.SendWait("^+8")
    return value
```
For some reason, this works, when the `SendKeys` operator does not.
